src/index.js

```diff
@@ -3,34 +3,52 @@ const legally = require("./legally");
 const analysis = require("./analysis");
 const remote = require("./remote");
 const clean = require("./options");
-const args = minimist(process.argv.filter(e => !/^\/.+$/.test(e)));
 
-// node on windows inserts extra into argv that we need to remove
-// could be a bug in minimist
-if (/^win/.test(process.platform)) args._ = args._.slice(2);
+async function RunNow()
+{
+  const args = minimist(process.argv.filter(e => !/^\/.+$/.test(e)));
+  var done;
+  var options;
+  var folder;
+  var licenses
 
-args.routes = args._;
+  // node on windows inserts extra into argv that we need to remove
+  // could be a bug in minimist
+  if (/^win/.test(process.platform))
+  {
+    args._ = args._.slice(2);
+  }
+  args.routes = args._;
 
-// Need to wait a bit before resolving
-// See: https://stackoverflow.com/a/50451612/938236
-var done = (function wait() {
-  if (!done) setTimeout(wait, 1000);
-})();
+  // Need to wait a bit before resolving
+  // See: https://stackoverflow.com/a/50451612/938236
+  done = (function wait() {
+    if (!done) setTimeout(wait, 1000);
+  })();
 
-(async () => {
   try {
-    const options = clean(args);
+    options = clean(args);
     console.log(
       `Working on "${options.routes.join(", ") ||
         "node_modules"}". It might take a while...`
     );
-    const folder = await remote(options.routes);
-    const licenses = await legally(folder);
+    folder = await remote(options.routes);
+    licenses = await legally(folder);
     await analysis(licenses, options);
   } catch (error) {
     console.error(error, args);
     throw error;
   } finally {
     done = true;
   }
+}
+
+module.exports = legally;
+(async () =>
+{
+  if (require.main.id === '.')
```

Comment:
Is it better to do `if (require.main === module) {`?

https://stackoverflow.com/a/45137096/938236

Reply:
I agree that would be better, but I could not get rollup to leave `module` alone. It kept turning it into a local variable that was just an empty object.

If you have any ideas I would prefer that since the NPM documentation says that is the supported way to do it.